src/middlewared/middlewared/plugins/pool.py

```diff
@@ -21,6 +21,29 @@
 logger = logging.getLogger(__name__)
 
 
+class Inheritable(EnumMixin, Attribute):
+    def __init__(self, *args, **kwargs):
+        self.value = kwargs.pop('value')
+        super(Inheritable, self).__init__(*args, **kwargs)
+
+    def clean(self, value):
+        if value == 'INHERIT':
+            return value
+
+        return self.value.clean(value)
+
+    def validate(self, value):
+        if value == 'INHERIT':
+            return
+
+        return self.value.validate(value)
+
+    def to_json_schema(self, parent=None):
+        schema = self.value.to_json_schema(parent)
+        schema['anyOf'] = [schema.pop('type'), {'type': 'string', 'enum': ['INHERIT']}]
```

Comment:
anyOf is ambiguous for OpenAPI, thats why I was insisting in making it a string or using null.

Reply:
That will result in:

["integer", {'type': 'string', 'enum': ['INHERIT']}]

which seems wrong